### ISSUE TITLE
Uncaught Error: ENOENT: no such file or directory, lstat '~/notes/.git/index.lock'

### ISSUE THREAD

**Issue by mlncn** [lexicalunit/atom-notes#59]
[Enter steps to reproduce:]

1. Have notes (automatically) committed to a git repository
2. Create and save a new note

**Atom**: 1.42.0 x64
**Electron**: 4.2.7
**OS**: Pop!_OS 19.10
**Thrown From**: [atom-notes](https://github.com/lexicalunit/atom-notes) package 1.20.2

### Stack Trace

Uncaught Error: ENOENT: no such file or directory, lstat '/home/mlncn/Projects/mlncn/notes/.git/index.lock'

```
At fs.js:115

Error: ENOENT: no such file or directory, lstat '/home/mlncn/Projects/mlncn/notes/.git/index.lock'
    at Proxy.realpathSync (fs.js:1435:7)
    at Proxy.fs.realpathSync (ELECTRON_ASAR.js:306:40)
    at NotesStore.addDocument (/packages/atom-notes/lib/notes-store.js:157:20)
    at /packages/atom-notes/lib/notes-store.js:99:14
    at PathWatcher.onNativeEvents (/usr/share/atom/resources/app/static/<embedded>:14:26758)
    at t.native.onDidChange.t (/usr/share/atom/resources/app/static/<embedded>:14:25494)
    at Function.simpleDispatch (/usr/share/atom/resources/app/static/<embedded>:11:1211760)
    at Emitter.emit (/usr/share/atom/resources/app/static/<embedded>:11:1213201)
    at NSFWNativeWatcher.onEvents (/usr/share/atom/resources/app/static/<embedded>:14:22740)
    at watcher.e (/usr/share/atom/resources/app/static/<embedded>:14:24389)
```

### Commands

```
  3x -0:32.2.0 blur (input.hidden-input)
     -0:28.6.0 vim-mode-plus:reset-normal-mode (input.hidden-input)
     -0:28.2.0 ex-mode:open (input.hidden-input)
  3x -0:28.2.0 blur (input.hidden-input)
     -0:27.3.0 core:confirm (input.hidden-input)
  6x -0:27.3.0 blur (input.hidden-input)
     -0:23.9.0 atom-notes:toggle (input.hidden-input)
  3x -0:23.9.0 blur (input.hidden-input)
     -0:22.6.0 date:date (input.hidden-input)
  2x -0:19.6.0 core:backspace (input.hidden-input)
     -0:15.8.0 core:confirm (input.hidden-input)
  4x -0:15.7.0 blur (input.hidden-input)
     -0:15 vim-mode-plus:activate-insert-mode (input.hidden-input)
     -0:13.2.0 markdown-writer:insert-new-line (input.hidden-input)
     -0:13.2.0 editor:newline (input.hidden-input)
  2x -0:00.2.0 blur (input.hidden-input)
```

### Non-Core Packages

```
atom-fuzzy-grep 0.17.2 
atom-ide-ui 0.13.0 
atom-notes 1.20.2 
atom-vim-like-tab 1.5.3 
busy-signal 2.0.1 
city-lights-syntax 1.1.8 
city-lights-ui 1.5.3 
date 2.6.0 
ex-mode 0.18.0 
file-templates 0.2.0 
github-url-resolver 0.1.1 
gitlab 0.1.3-7 
ide-css 0.3.3 
ide-html 0.6.0 
ide-json 0.2.1 
ide-php 0.7.18 
ide-python 1.5.0 
ide-yaml 0.6.0 
intentions 1.1.5 
linter 2.3.1 
linter-php 1.6.1 
linter-ui-default 1.8.1 
markdown-pdf 2.2.0 
markdown-writer 2.11.9 
open-terminal-here 2.3.2 
paste-table 0.1.0 
path-copy 0.14.0 
php-cs-fixer 4.1.1 
php-twig 4.0.0 
project-folder 1.5.0 
project-manager 3.3.8 
rst-preview-pandoc 0.1.12 
save-as-heading 1.3.0 
snippet-generator 0.3.2 
teletype 0.13.4 
tidy-tabs 0.4.0 
undo-tree 1.0.0 
vertical-tabs-plus 2.4.0 
vim-mode-plus 1.36.4 
Writer 0.1.0 
```

**Comment by lexicalunit** [comment]
You probably want to make sure that `atom-notes` ignores the `.git` directory so that it doesn't try to index any transient files that git may generate as a matter of course. Working on a fix...

**Comment by mlncn** [comment]
Thank you so much Amy!  That's fast work!  One thing i should have mentioned is i have had this setup for a while, so i suspect something changed in Atom Notes to make this an issue, but— a fix is a fix!

**Comment by lexicalunit** [comment]
Just published a new minor version. Give version 1.21.0 a try and let me know if that resolves the issue for you 🎉

**Comment by oschwery** [comment]
Hi, I have the same problem (though only on mac, I believe, not on windows). I'm on 1.21.1 (so post fix) and .git is set to be ignored.

Any ideas why it still seems to track those when I commit changes?

Thanks!

___________

[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.44.0 x64
**Electron**: 4.2.7
**OS**: Mac OS X 10.15.3
**Thrown From**: [atom-notes](https://github.com/lexicalunit/atom-notes) package 1.21.1


### Stack Trace

Uncaught Error: ENOENT: no such file or directory, lstat '/Users/oschwery/Documents/nvAtom/.git/refs/heads/master.lock'

```
At fs.js:115

Error: ENOENT: no such file or directory, lstat '/Users/oschwery/Documents/nvAtom/.git/refs/heads/master.lock'
    at Proxy.realpathSync (fs.js:1435:7)
    at Proxy.fs.realpathSync (ELECTRON_ASAR.js:306:40)
    at NotesStore.addDocument (/packages/atom-notes/lib/notes-store.js:157:20)
    at /packages/atom-notes/lib/notes-store.js:99:14
    at PathWatcher.onNativeEvents (/Users/oschwery/Applications/Atom.app/Contents/Resources/app/static/<embedded>:14:26758)
    at t.native.onDidChange.t (/Users/oschwery/Applications/Atom.app/Contents/Resources/app/static/<embedded>:14:25494)
    at Function.simpleDispatch (/Users/oschwery/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:1212986)
    at Emitter.emit (/Users/oschwery/Applications/Atom.app/Contents/Resources/app/static/<embedded>:11:1214427)
    at NSFWNativeWatcher.onEvents (/Users/oschwery/Applications/Atom.app/Contents/Resources/app/static/<embedded>:14:22740)
    at watcher.e (/Users/oschwery/Applications/Atom.app/Contents/Resources/app/static/<embedded>:14:24389)
```

### Commands

```
 11x -1:31.2.0 core:move-left (input.hidden-input)
  5x -1:09.8.0 core:backspace (input.hidden-input)
     -1:03.4.0 editor:newline (input.hidden-input)
```

### Non-Core Packages

```
atom-notes 1.21.1 
cowabunga 0.2.0 
dark-petroleum-syntax 0.1.3 
emerald-syntax 1.1.5 
glacier-syntax 0.0.4 
gotham-syntax 0.2.0 
language-atom-notes 0.4.1 
language-r 0.4.2 
language-revbayes 0.0.0 
minimap 4.29.9 
minimap-codeglance 0.4.7 
minimap-cursorline 0.2.0 
minimap-find-and-replace 4.5.2 
minimap-selection 4.5.0 
r-exec 0.5.0 
teletype 0.13.4 
wakatime 7.1.1 
Watchmenesque-Dark 0.14.0 
```

**Comment by lexicalunit** [comment]
I think I see what's happening. Thanks for the bug report @oschwery!

**Comment by oschwery** [comment]
BTW, I noticed you released 1.22.0, so I updated to that, and so far the issue has disappeared for now. Thanks for the quick fix!
[and thanks for the package to begin with. I spent quite some time for something that's just like nvALT, but cross-platform, and this is by far the closest, yet most light-weight option to find! 😃]

**Comment by lexicalunit** [comment]
Oh sorry, I forgot to update the issue to let you know I updated it to fix the issue!
